src/libraries/System.Configuration.ConfigurationManager/tests/System/Configuration/ConfigurationPathTests.cs

```diff
@@ -0,0 +1,92 @@
+            }, configFilePath, expectedSettingValue).Dispose();
+        }
+
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void CustomAppConfigIsUsedWhenSpecifiedAsAbsolutePath()
+        {
+            const string SettingName = "test_CustomAppConfigIsUsedWhenSpecified";
+            string expectedSettingValue = Guid.NewGuid().ToString();
+            string configFilePath = Path.GetFullPath(CreateAppConfigFileWithSetting(SettingName, expectedSettingValue));
+
+            RemoteExecutor.Invoke((string configFilePath, string expectedSettingValue) => {
+                AppDomain.CurrentDomain.SetData(ConfigName, configFilePath);
+                Assert.Equal(expectedSettingValue, ConfigurationManager.AppSettings[SettingName]);
+            }, configFilePath, expectedSettingValue).Dispose();
+        }
+
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void CustomAppConfigIsUsedWhenSpecifiedAsAbsoluteUri()
+        {
+            const string SettingName = "test_CustomAppConfigIsUsedWhenSpecified";
+            string expectedSettingValue = Guid.NewGuid().ToString();
+            string configFilePath = new Uri(Path.GetFullPath(CreateAppConfigFileWithSetting(SettingName, expectedSettingValue))).ToString();
+
+            RemoteExecutor.Invoke((string configFilePath, string expectedSettingValue) => {
+                AppDomain.CurrentDomain.SetData(ConfigName, configFilePath);
+                Assert.Equal(expectedSettingValue, ConfigurationManager.AppSettings[SettingName]);
+            }, configFilePath, expectedSettingValue).Dispose();
+        }
+
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void NoErrorWhenCustomAppConfigIsSpecifiedAndItDoesNotExist()
+        {
+            RemoteExecutor.Invoke(() =>
+            {
+                AppDomain.CurrentDomain.SetData(ConfigName, "non-existing-file.config");
+                Assert.Null(ConfigurationManager.AppSettings["AnySetting"]);
+            }).Dispose();
+        }
+
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void MalformedAppConfigCausesException()
+        {
+            const string SettingName = "AnySetting";
+
+            // Following will cause malformed config file
+            string configFilePath = CreateAppConfigFileWithSetting(SettingName, "\"");
+
+            RemoteExecutor.Invoke((string configFilePath) => {
+                AppDomain.CurrentDomain.SetData(ConfigName, configFilePath);
+                Assert.Throws<ConfigurationErrorsException>(() => ConfigurationManager.AppSettings[SettingName]);
+            }, configFilePath).Dispose();
+        }
+
+        private static string CreateAppConfigFileWithSetting(string key, string rawUnquotedValue)
+        {
+            string fileName = Path.GetRandomFileName() + ".config";
```

Comment:
These files will leak. it might be easiest to derive from FileCleanupTestBase and call GetTestFileName().

Reply:
this gets a bit tricky when I want to test @JeremyKuhne's scenario https://github.com/dotnet/runtime/pull/56748#discussion_r681377591

I'll use that for everything but related test cases

Reply:
I have ended up modifying constructor of FileCleanupTestBase so that it allows to pickup something else that TempDirectory which in here I chose to be AppDomain.BaseDirectory - TempDirectory doesn't allow me to test relative paths correctly